projects/ruby/fuzz_ruby_gems.c

```diff
@@ -0,0 +1,400 @@
+/* Copyright 2022 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "init_ruby_load_paths.h"
+#include <ruby/ruby.h>
+#include <unistd.h>
+
+#define ARRAYSIZE(x) (sizeof(x) / sizeof(x[0]))
+
+enum RubyDataType { RDT_CString };
+
+struct TargetFunction {
+  const char *module_;
+  const char *cls_;
+  const char *name_;
+  VALUE module_obj_;
+  VALUE cls_obj_;
```

Comment:
If these VALUEs are going to be stored as globals in C they need to be marked or registered through `rb_global_variable`, otherwise they're allowed to be moved in compaction (or freed if Ruby loses reference to them)

Reply:
Thanks, I added two calls to `rb_global_variable`.